tests/component-tests/textfieldSlider/js/TextfieldSliderTests.js

```diff
@@ -21,86 +21,190 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+                    "method": "val"
+                },
+                "setSliderValue": {
+                    "this": "{that}.slider.container",
+                    "method": "val",
+                    args: ["{arguments}.0"]
+                }
+            }
+        });
+
+        fluid.defaults("fluid.tests.textfieldSlider.jQueryUI", {
+            gradeNames: ["fluid.textfieldSlider"],
+            invokers: {
+                "getSliderValue": {
+                    "this": "{that}.slider.slider",
+                    "method": "slider",
+                    args: ["value"]
+                },
+                "setSliderValue": {
+                    "this": "{that}.slider",
+                    "method": "setSliderValue",
+                    args: ["{arguments}.0"]
+                }
+            }
+        });
+
+        fluid.tests.textfieldSlider.createTextfieldSliderNativeHTML = function (options) {
+            // Make sure we're using the native widget
+            fluid.contextAware.makeChecks({
+                "fluid.prefsWidgetType": {
+                    value: "nativeHTML"
+                }
+            });
+            return fluid.tests.textfieldSlider.nativeHTML(".fl-textfield-slider-native", options);
         };
 
-        jqUnit.test("Test Init", function () {
+        fluid.tests.textfieldSlider.createTextfieldSliderJQueryUI = function (options) {
+            // Override default native widget to use jQuery slider instead
+            fluid.contextAware.makeChecks({
+                "fluid.prefsWidgetType": {
+                    value: "jQueryUI"
+                }
+            });
+
+            return fluid.tests.textfieldSlider.jQueryUI(".fl-textfield-slider-jQuery", options);
+        };
+
+        fluid.tests.textfieldSlider.testCommonInit = function (textfieldSlider) {
+            var that = textfieldSlider;
+
+            jqUnit.assertEquals("Slider value is set to input value", 15, +textfieldSlider.getSliderValue());
+            jqUnit.assertEquals("Textfield value is set", 15, +that.locate("textfield").val());
+            jqUnit.assertEquals("The model should be set", 15, that.model.value);
+            jqUnit.assertEquals("Min should be the default", 0, that.options.range.min);
+            jqUnit.assertEquals("Max should be the default", 100, that.options.range.max);
+        };
+
+        jqUnit.test("Test Init (native HTML slider)", function () {
+            jqUnit.expect(8);
+            var that = fluid.tests.textfieldSlider.createTextfieldSliderNativeHTML({model: {value: 15}});
+
+            fluid.tests.textfieldSlider.testCommonInit(that);
+
+            var slider = that.locate("slider");
+
+            // Check range slider attributes
+            jqUnit.assertEquals("The value now should be 15", 15, +that.getSliderValue());
+            jqUnit.assertEquals("The max should be 100", 100, +slider.attr("max"));
+            jqUnit.assertEquals("The min should be 0", 0, +slider.attr("min"));
```

Comment:
There needs to be cleanup of the context at the end of these tests with a corresponding call to fluid.contextAware.forgetChecks